code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,42 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("failed to generate Deepmaint - no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		var/success = deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		if(success)
+			log_and_message_admins("has generated a DeepmaintRooms map.")
+			break
+		else
+			log_and_message_admins("has unsuccessfully attempted to generate a DeepmaintRooms map.")
```

Comment:
Would these not make more sense outside of the loop, to avoid spam in case of concurrent failures? Would also give a reason to use the `success` var defined in the previous comment

Reply:
Fair shout, but as-is it gives you an indication of how badly messed up it is (normally, it should work on first try; if you see multiple failures, something has gone off the rails.

Reply:
Eh, do we really need to spam all admins with it though? Each call is also an I/O write to file iirc. Could you not make a counter var if you want to see how badly it screwed up, and count the amount of attempts and log that instead?